cmd/allowPrivilegeEscalation.go

```diff
@@ -49,10 +49,12 @@ func checkAllowPrivilegeEscalation(container Container, result *Result) {
 func auditAllowPrivilegeEscalation(resource k8sRuntime.Object) (results []Result) {
 	for _, container := range getContainers(resource) {
 		result := newResultFromResource(resource)
-		checkAllowPrivilegeEscalation(container, &result)
-		if len(result.Occurrences) > 0 {
-			results = append(results, result)
-			break
+		if result != nil {
```

Comment:
Instead of checking `result != nil` it would be better to change the function newResultFromResource to return an err explicit stating that the particular resource type is not supported. This helps in understanding what exactly happened instead of silently ignoring it with `result != nil`

Reply:
Would something akin to this work ?

```result, err := newResultFromResource(resource)
		if err == ErrResourceTypeNotSupported {
			continue
		} else if err != nil {
			panic(err)
		}```